src/lib/SVGMap/SVGMap.svelte

```diff
@@ -0,0 +1,190 @@
+<script>
+  import { reducedMotion } from "../stores";
+  import { readable, writable } from "svelte/store";
+  import { setContext, onMount } from "svelte";
+  import { geoAlbersUsa } from "d3-geo";
+  import { zoom, zoomIdentity } from "d3-zoom";
+  import { select } from "d3-selection";
+  import ZoomControls from "./ZoomControls.svelte";
+
+  /*
+   * An array of geojson features to be displayed on the map. The map will scale the projection to fit this set of features.
+   */
+  export let features;
+  export let projection = geoAlbersUsa;
+  export let zoomable = false;
+
+  /*
+   * Where the zoom control UI should be positioned.
+   * @type {"bottom-left" | "bottom-right" | "top-left" | "top-right"}
+   */
+  export let controlPosition = "bottom-right";
+  export let height = 800;
+
+  /*
+   * Optional aria role string to be applied to SVG container. By default, the SVG is hidden from the accessiblity tree. If you add an ariaRole here, any layers should also be given an ariaRole.
+   * @type { string } [ariaRole = undefined]
+   */
+  export let ariaRole = undefined;
+
+  /*
+   * Optional aria label string to be applied to SVG container. By default, the SVG is hidden from the accessiblity tree and should include a descriptive label. If you add an ariaRole this property can be left undefined;
+   * @type { string } [ariaRole = undefined]
+   */
+  export let ariaLabel = undefined;
+
+  /**
+   * Whether or not the map should zoom when scroll wheel is used on map.
+   * @type {"no" | "yes" | "ctrl"}
+   */
+  export let scrollWheel = "no";
+
+  // create stores of map global settings to add to context
+  $: featuresStore = readable(features);
+  let width = 500;
+
+  // size to fit projection to
+  $: fitSizeRange = [width, height];
+
+  // setup the scaled projection function
+  $: projectionFn = projection().fitSize(fitSizeRange, {
+    type: "FeatureCollection",
+    features: features
+  });
+
+  // set up a store to hold the scaled projection function
+  const projectionStore = writable(projectionFn);
+  // update the store any time the projection function updates
+  $: projectionStore.set(projectionFn);
+
+  // initialize a transform store in case zoom is turned on
+  // this will be where we store the output of the d3-zoom behaviour and broadcast it
+  let transformStore = writable(zoomIdentity);
+
+  // will hold d3.zoom instance if zoomable
+  let mapZoom;
+
+  // will hold svg selection if zoomable
+  let svgSelection;
+
+  // add global stores to context
+  $: setContext("map", {
+    projection: projectionStore,
+    features: featuresStore,
+    transform: transformStore
+  });
+
+  function setupZoom(el) {
+    // create d3 selection of the svg
+    svgSelection = select(el.querySelector("svg"));
+    // create zoom instance
+    mapZoom = zoom()
+      .scaleExtent([1, 8])
+      .filter((event) => {
+        if (scrollWheel === "ctrl") {
```

Comment:
consider mac users with `alt` key as well (maybe it's both ctrl or alt if ctrl is selected?) - related: mapbox's [cooperative gestures](https://docs.mapbox.com/mapbox-gl-js/example/cooperative-gestures/)

Reply:
ah yeah interesting. i feel like just the `ctrl` approach is a wider standard (Datawrapper works with just `ctrl`), but i do kinda like the `command` option for mac users...probably no downside to allowing both

Reply:
ok, `metaKey` is now supported in addition to `ctrlKey` for this.